### DISCUSSION TITLE
Homework web app

### DISCUSSION THREAD

**Discussion by MixtarMaster** [a2br/ecoledirecte.js#155] [Show and tell]
HEYOO!!!

I just wanted to show you all the small web app I created with this repo!

Gives you your homework and "contenue de séance" from EcoleDirecte and mails it to you!!

What do you think?

[Here](https://github.com/MixtarMaster/GetMeMyHomework) it is! I'm open to feedback!

**Comment by JsonLinesCode** [comment]
Looks super nice ! 🎉

**Reply by MixtarMaster** [reply]
Thanks!

**Comment by a2br** [comment]
C'est super! Je travaille aussi sur un projet qui étend EcoleDirecte. D'ailleurs, à ses débuts, ed.js n'était qu'une librairie interne à ce projet...

**Reply by a2br** [reply]
Attends... T'es sûr que ton code marche? 🤔

**Reply by a2br** [reply]
J'ai ouvert une PR :)

**Reply by MixtarMaster** [reply]
Pourquoi il marcherais pas, j'ai eu aucun problème? 

J'ai plusieurs question sur la PR:
Le commit "add type check for intellisense", pourquoi l'avoir mis? Juste éviter les erreurs?

Pourquoi enlever les var? Quelle est la différence avec let?

Pour tes propositions pour réparer, je comprends pas pourquoi les mettre puisque tout marche pour moi.

Dernière question: puis-je choisir quelles commit intégrer ou faut-il tout mettre si on en veut une partie?

**Reply by a2br** [reply]
J'ai mis le check pour le type pour que intellisense aide sur les méthodes élèves,

Les let sont grosso modo la version plus récente des var, c'est meilleure pratique de les utiliser

Et les promesses entourées de fonctions async sont... Des fonctions, pas la valeur qu'ils retournent, donc il y a aucune chance que ça marche (sinon je loupe un truc)

T'es pas obligé d'accepter la PR ;) jsp comment on accepté qu'un commit par contre. J'ai une idée mais c pas faisable sur le web je crois

**Reply by MixtarMaster** [reply]
Ok, merci!

Je t'assure que ça marche nickel, après peut-être que j'utilise un bug ou une méthode autre que tu avait pas prévu?
J'utilise async vraiment juste pour pouvoir l'utiliser comme une fonction malgré les await. 

Je tacherai aussi de faire moins de "spaghetti code" et de le rendre plus joli hehe.

En tout cas merci pour les conseil!

**Reply by MixtarMaster** [reply]
Très intéressant: j'ai ajouté la ligne ``if (account.type !== "student") throw new Error("Not a student");`` et ça ma sortie l'erreur "not a student" alors que sans la ligne, tout marche très bien

**Reply by MixtarMaster** [reply]
Au fait, à quoi sert : .prettierrc.json ?

**Reply by a2br** [reply]
C'est normal... fais un console.log de account.type. Normalement tu devrais log une promesse. C'est pas bon, il faut l'attendre, la promesse.
.prettierrc.json est le fichier de config de prettier. Quand installé dans VSCode & que tu save, il va reformater ton code d'après les règles du fichier pour que ce soit comme du veux.

**Reply by MixtarMaster** [reply]
Ok ok, j'ai encore un peu de mal avec tout ce qui est async et promesse..

**Reply by a2br** [reply]
Oui, ça m'avait aussi pris du temps ;) je crois que je l'ai déjà dit, mais utiliser TypeScript aide beaucoup. Peut-être que [ce lien](https://www.freecodecamp.org/news/javascript-es6-promises-for-beginners-resolve-reject-and-chaining-explained/) aidera

**Comment by JsonLinesCode** [comment]
C'est un fichier de configuration pour le formatter Prettier qui permet à
tous les développeurs d'utiliser le même formatage (à condition qu'ils
utilisent Prettier) en développant ce projet. Ça évite notamment des pull
requests avec des fichiers d'apparence plein de modifications mais qui en
fait ne sont que des changements de formatage automatiquement effectuées
par les formatters comme... Prettier.
